refactor(service-worker): improve localhost detection in scheduleInit…

SkyZeroZx opened the pull request:
Replace simple string search with comprehensive regex pattern that detects:
- localhost hostname
- IPv6 loopback address [::1]  
- IPv4 loopback range 127.x.x.x

Extract logic into dedicated isLocalhost() method for better code organization
and remove existing TODO comment.

JoostK commented on packages/service-worker/worker/src/driver.ts:
```diff
@@ -931,13 +931,18 @@ export class Driver implements Debuggable, UpdateSource {
     this.idle.schedule(`initialization(${appVersion.manifestHash})`, initialize);
   }
 
+  private isLocalhost(): boolean {
+    return /(localhost|\[::1\]|127(?:\.\d{1,3}){3})(:\d+)?\/?$/.test(this.scope.registration.scope);
+  }
```
I'd suggest pulling this out into a top-level function that takes the scope as parameter, such that you can add tests for this logic.

Also the regex can be adjusted to not have any capturing groups:


```suggestion
  private isLocalhost(): boolean {
    return /(?:localhost|\[::1\]|127(?:\.\d{1,3}){3})(?::\d+)?\/?$/.test(this.scope.registration.scope);
  }
```

Q should the localhost case be prefixed with something like `\W` to require a non-word in front of it?

SkyZeroZx replied:
I think so, I just updated with some improvements and tests.

SkyZeroZx committed:
refactor(service-worker): improve localhost detection in scheduleInitialization

Replaces ad hoc localhost check with a dedicated method using a regular expression for more reliable detection

thePunderWoman reviewed (approved):
LGTM!
